src/deadline/client/ui/widgets/host_requirements_tab.py

```diff
@@ -859,19 +864,27 @@ def has_input(self) -> bool:
         return PLACEHOLDER_TEXT != self.currentText()
 
 
-class OptionalSpinBox(QSpinBox):
+class OptionalSpinBox(QDoubleSpinBox):
     """
-    A custom QSpinBox that set min - 1 value as "-" to represent value not set.
+    A custom QDoubleSpinBox that set min - 1 value as "-" to represent value not set.
     """
 
-    def __init__(self, min: int = MIN_INT_VALUE, max: int = MAX_INT_VALUE, parent=None) -> None:
+    def __init__(
+        self,
+        min: int = MIN_INT_VALUE,
+        max: int = MAX_INT_VALUE,
+        decimal: int = DECIMAL_VALUE,
+        parent=None,
+    ) -> None:
```

Comment:
OptionalSpinBox appears to be a public class and this change will break anyone only using positional args, so I think we'll either have to mark this as a breaking change or add decimal after parent.

Other option is to still make this a breaking change, but only accept keyword args: 

```python
def func(self, *, key="word", only="args"):
```

Reply:
I could make it non-breaking by not including decimal in the constructor, defaulting it to `DECIMAL_VALUE` and then using `setDecimals()` where I need it changed?

Reply:
Yea that works as well